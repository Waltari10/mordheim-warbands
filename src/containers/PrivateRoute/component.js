import React, { memo } from 'react';
import { Route, Redirect } from 'react-router-dom';
import CircularProgress from '@material-ui/core/CircularProgress';


const PrivateRoute = memo(({ component: Component, isLoading, isAuthorized, ...rest }) => {

  if (isLoading) {
    return <CircularProgress />;
  }

  if (!isAuthorized && !isLoading) {
    return (
      <Redirect
        to={{
          pathname: '/login',
          state: { from: rest.location },
        }}
      />
    );
  }

  if (Component) {
    return (
      <Route
        {...rest}
        render={props =>(<Component {...props} />)}
      />
    );
  }

  if (rest.render) {
    return (
      <Route
        {...rest}
        render={rest.render}
      />
    );
  }


});

export default PrivateRoute;