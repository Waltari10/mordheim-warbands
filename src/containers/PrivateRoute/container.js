import { connect } from 'react-redux';

import Component from './component';

function mapStateToProps(state) {

  const isAuthorized = !!state.login.user;

  return {
    isAuthorized,
  };
}


export default connect(
  mapStateToProps,
  {}
)(Component);
