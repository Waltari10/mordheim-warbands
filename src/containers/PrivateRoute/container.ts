import { connect } from 'react-redux';

import Component from './component';

export interface StateProps {
  isAuthorized?: boolean;
  isLoading?: boolean;
}

function mapStateToProps(state) {

  const isAuthorized = !!state.user.user;
  const isLoading = !!state.user.isLoading;

  return {
    isAuthorized,
    isLoading,
  };
}


export default connect(
  mapStateToProps,
  {}
)(Component);
