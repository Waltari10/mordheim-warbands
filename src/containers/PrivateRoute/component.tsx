import React, { memo } from 'react';
import { Route, Redirect } from 'react-router-dom';
import CircularProgress from '@material-ui/core/CircularProgress';
import { StateProps } from './container';

interface Props {
  component?: any;
  location?: any;
  render?: any;
  path?: any;
}


const PrivateRoute = memo(
  (
    {
      component,
      isLoading,
      isAuthorized,
      ...rest
    }: StateProps & Props
  ) => {

    const Component = component;

    if (isLoading) {
      return <CircularProgress />;
    }

    if (!isAuthorized && !isLoading) {
      return (
        <Redirect
          to={{
            pathname: '/login',
            state: { from: rest.location },
          }}
        />
      );
    }

    if (Component) {
      return (
        <Route
          {...rest}
          render={props => (<Component {...props} />)}
        />
      );
    }

    if (rest.render) {
      return (
        <Route
          {...rest}
          render={rest.render}
        />
      );
    }

    return null;


  });

export default PrivateRoute;