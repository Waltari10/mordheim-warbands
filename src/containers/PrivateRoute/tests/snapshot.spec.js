import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import PrivateRoute from '../component';

const Test = () => {
  return (<div></div>);
};


describe('PrivateRoute', () => {
  it('renders PrivateRoute', () => {
    const tree = renderer
      .create(
        <Router>
          <PrivateRoute
            isAuthorized
            component={Test}
          />
        </Router>
      )
      .toJSON();


    expect(tree).toMatchSnapshot();
  });
});