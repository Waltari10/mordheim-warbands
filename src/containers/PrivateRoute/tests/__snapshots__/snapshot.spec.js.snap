// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute renders PrivateRoute 1`] = `<div />`;
